tests/unit/test_session.py

```diff
@@ -549,6 +549,23 @@ def test_cred_provider_called_when_partial_creds_provided(self):
                 aws_secret_access_key='foo',
             )
 
+    @mock.patch('botocore.client.ClientCreator')
+    def test_credential_params(self, client_creator):
```

Comment:
I couldn't find a test that explicitly checked if each of the credential values that can be statically provided to `create_client` were properly set on the credentials, so I added one. We would need something like this for account ID anyway.